packages/psss-server/src/routes/reports/SubmitCountryWeeklyReportRoute.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { RespondingError } from '@tupaia/utils';
+import { Route } from '../Route';
+
+const SURVEY_CODE = 'PSSS_WNR';
+
+type WeeklyReportAnswers = {
+  PSSS_Sites: number;
+  PSSS_Sites_Reported: number;
+  PSSS_AFR_Cases: number;
+  PSSS_DIA_Cases: number;
+  PSSS_ILI_Cases: number;
+  PSSS_PF_Cases: number;
+  PSSS_DLI_Cases: number;
+};
+
+export class SubmitCountryWeeklyReportRoute extends Route {
+  async buildResponse() {
+    const { week } = this.req.query;
+    const { organisationUnitCode } = this.req.params;
+    const answers = mapReqBodyToAnswers(this.req.body);
+
+    const existingSurveyResponse = await this.meditrakConnection?.findSurveyResponse(
+      SURVEY_CODE,
+      organisationUnitCode,
+      week,
+    );
+
+    if (existingSurveyResponse) {
+      return this.meditrakConnection?.updateSurveyResponse(existingSurveyResponse, answers);
+    }
+
+    return this.meditrakConnection?.createSurveyResponse(
+      SURVEY_CODE,
+      organisationUnitCode,
+      week,
+      answers,
+    );
+  }
+}
+
+const mapReqBodyToAnswers = (body: Record<string, unknown>): WeeklyReportAnswers => {
+  const {
+    Sites: sites,
+    'Sites Reported': sitesReported,
+    AFR: afr,
+    DIA: dia,
+    ILI: ili,
+    PF: pf,
+    DLI: dli,
+  } = body;
+  return {
+    PSSS_Sites: validateIsNumber(sites),
+    PSSS_Sites_Reported: validateIsNumber(sitesReported),
+    PSSS_AFR_Cases: validateIsNumber(afr),
+    PSSS_DIA_Cases: validateIsNumber(dia),
+    PSSS_ILI_Cases: validateIsNumber(ili),
+    PSSS_PF_Cases: validateIsNumber(pf),
+    PSSS_DLI_Cases: validateIsNumber(dli),
+  };
+};
+
+const validateIsNumber = (value: unknown): number => {
+  if (typeof value !== 'number') {
+    throw new RespondingError(`Cannot confirm weekly data: invalid data`, 500);
+  }
+
+  return value;
+};
```

Comment:
Really feels like this should be shared across the two endpoints, rather than duplicated

Either way: copy paste error on the error thrown, should be `Cannot save weekly report`

Reply:
Ah good catch on the error string and you make a good point about shared code!

I think ideally we'd have a common `utils` package in typescript as I really want to share the code that exists in the `indicators` package which validates objects (`src/builders/helpers.ts#validateConfig`)

What I might do for now is create a similar function in a `utils` directory within psss-server, and create a techdebt ticket to either consolidate such functions into a `@tupaia/ts-utils` package, or for us to convert `@tupaia/utils` to typescript?